erpnext/stock/doctype/quality_inspection/quality_inspection.js

```diff
@@ -18,6 +47,15 @@ frappe.ui.form.on("Quality Inspection", {
 					refresh_field(['quality_inspection_template', 'readings']);
 				}
 			});
+
+		}
+	},
+	reference_type: function (frm) {
+		if (frm.doc.reference_type == "Purchase Invoice" || frm.doc.reference_type == "Purchase Receipt") {
+			frm.toggle_reqd('reference_name', true);
+		}
+		else {
+			frm.toggle_reqd('reference_name', false);
 		}
 	},
```

Comment:
Why are we doing this exactly? It shouldn't be mandatory, if it's not set, just don't fetch the purchase details below